engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java

```diff
@@ -60,27 +60,27 @@ public DbDeploymentState(
   public void addPendingDeploymentDistribution(final long deploymentKey, final int partition) {
     this.deploymentKey.wrapLong(deploymentKey);
     partitionKey.wrapInt(partition);
-    pendingDeploymentColumnFamily.put(deploymentPartitionKey, DbNil.INSTANCE);
+    pendingDeploymentColumnFamily.insert(deploymentPartitionKey, DbNil.INSTANCE);
   }
 
   @Override
   public void removePendingDeploymentDistribution(final long deploymentKey, final int partition) {
     this.deploymentKey.wrapLong(deploymentKey);
     partitionKey.wrapInt(partition);
-    pendingDeploymentColumnFamily.delete(deploymentPartitionKey);
+    pendingDeploymentColumnFamily.deleteIfExists(deploymentPartitionKey);
```

Comment:
Seems to work, except for the helpfully named `shouldRemovePendingDeploymentIdempotent` test that tests that removing a pending deployment twice works. I don't know _why_ that test exists though.

Reply:
@korthout Any idea about this test and whether it is important?

Reply:
When we implemented the distribution we added this test https://github.com/camunda-cloud/zeebe/commit/714ca6bb6cb40bb69fca809395aab67bfd8863a5#diff-c5de5f2a4beee2bb8434aa9fd91bd598357ee653f7323eb169bbf003a4c3b302 because we wanted that the complete behaves idempotent https://github.com/camunda-cloud/zeebe/commit/714ca6bb6cb40bb69fca809395aab67bfd8863a5#diff-6fd0cf018a0856491734c8c63b66cf53f54c4ee95496f79cbd4bdade2f8fcf6b

later we removed the idempotency see here https://github.com/camunda-cloud/zeebe/commit/befebf357e679833ebc3d514e40fe5f65d3cfef7#diff-6fd0cf018a0856491734c8c63b66cf53f54c4ee95496f79cbd4bdade2f8fcf6b

So i think the test can be removed

Reply:
Thanks for looking into it @Zelldon, I'll remove the test then and use `deleteExisting` :+1: